tests/test_client.py

```diff
@@ -928,6 +929,190 @@ def test_asynchronous_delivery_failures_are_logged(self):
             ANY
         )
 
+    def test_chained_exceptions_with_explicit_cause(self):
+        self.client.notify(fixtures.exception_with_explicit_cause)
+
+        assert self.sent_report_count == 1
+
+        payload = self.server.received[0]['json_body']
+        exceptions = payload['events'][0]['exceptions']
+
+        assert len(exceptions) == 3
+
+        assert exceptions[0]['message'] == 'a'
+        assert exceptions[0]['errorClass'] == 'NameError'
+
+        # TODO: this stacktrace is generated using 'sys.exc_info()', so starts
+        #       where we construct the Event
+        #       switching to 'exception.__traceback__' should improve this as
+        #       it will use the traceback from 'exception_with_explicit_cause'
```

Comment:
Specifically I'm talking about changing this line in the Event class:

https://github.com/bugsnag/bugsnag-python/blob/67311be1b72c4c5023f2d44efe2a63b0e2f68dd8/bugsnag/event.py#L84

I think we're using `sys.exc_info` for historical reasons, as Python 3.0 added `exception.__traceback__` but I've not really investigated this yet. I'll look into this separately as it's not really related to caused by exceptions

Reply:
almost :100:% sure that your hunch is correct